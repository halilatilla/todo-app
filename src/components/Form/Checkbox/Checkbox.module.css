.wrapper {
  @apply flex cursor-pointer items-center space-x-3;
}

.checkbox {
  @apply h-6 w-6 appearance-none rounded !bg-white;
  transform: translateY(-0.075em);
  display: grid;
  place-content: center;
}

.label {
  @apply text-sm font-medium text-gray-600;
}

.checkbox::before {
  content: '';
  width: 14px;
  height: 14px;
  clip-path: polygon(14% 44%, 0 65%, 50% 100%, 100% 16%, 80% 0%, 43% 70%);
  transform: scale(0);
  transform-origin: bottom left;
  transition: 120ms transform ease-in-out;
  @apply bg-green-600;
}

.checkbox:checked::before {
  transform: scale(1);
}
