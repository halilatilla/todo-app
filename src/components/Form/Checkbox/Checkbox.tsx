import { FC } from 'react'
import classnames from 'classnames'

import styles from './Checkbox.module.css'

interface Props {
  label?: string
  className?: string
  classNameCheckbox?: string
  classNameLabel?: string
  checked: boolean
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
}

const Checkbox: FC<Props> = ({ label, className, classNameCheckbox, classNameLabel, ...rest }) => {
  return (
    <label className={classnames(styles.wrapper, className)}>
      <input type="checkbox" className={classnames(styles.checkbox, classNameCheckbox)} {...rest} />
      {label && <span className={classnames(styles.label, classNameLabel)}>{label}</span>}
    </label>
  )
}

export default Checkbox
