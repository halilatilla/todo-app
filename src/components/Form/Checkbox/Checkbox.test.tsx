import { render, screen } from '@testing-library/react'

import Checkbox from './Checkbox'

describe('Checkbox', () => {
  const DATA = {
    onChange: () => {},
  }

  it('renders correctly', () => {
    const { container } = render(<Checkbox {...DATA} />)
    expect(container).toMatchSnapshot()
  })

  it('renders Checkbox', () => {
    render(<Checkbox {...DATA} />)

    const checkbox = screen.getByRole('checkbox')

    expect(checkbox).toBeInTheDocument()
  })
})
