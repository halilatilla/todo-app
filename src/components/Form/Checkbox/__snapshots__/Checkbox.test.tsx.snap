// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<div>
  <label
    class="wrapper"
  >
    <input
      class="checkbox"
      type="checkbox"
    />
  </label>
</div>
`;
